website/setupTypedoc.ts

```diff
@@ -0,0 +1,33 @@
+const fs = require('fs');
+const path = require('path');
+
+// Define the directory where the Typedoc HTML files are located
+const typedocDir = path.join(__dirname, 'static/typedoc_output');
+
+// Define a recursive function to find all HTML files in a directory
+function findHtmlFiles(dir, fileList = []) {
+  const files = fs.readdirSync(dir);
+
+  files.forEach(file => {
+    if (fs.statSync(path.join(dir, file)).isDirectory()) {
+      fileList = findHtmlFiles(path.join(dir, file), fileList);
+    } else if (file.endsWith('.html')) {
+      fileList.push(path.join(dir, file));
+    }
+  });
+
+  return fileList;
+}
+
+// Find all HTML files in the Typedoc directory
+const htmlFiles = findHtmlFiles(typedocDir);
+
+// Go through each HTML file and add the target="_parent" attribute to the external links
```

Comment:
why? Can you guide me to references to read on this workaround?

Reply:
Basically if we were to host typedoc docs as static content, we would lose docusaurus navigation. Hence someone clicking on typedoc, would have to use the back arrow to go back to the main doc website page, or to move to other pages. To solve that, I created a new docusaurus page and embedded the root of the typedoc generated documentation using an iframe. By default iframes tend to navigate within the iframe itself on links, unless you specify target="_parent" -> TLDR this allows to have nagivation even on typedoc and to make links within the iframe work

Reply:
another option was to use some pluging from docusaurus to compile typedoc as markdown, but it would not look as pretty as it does in html